Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
     {
         if ($event instanceof AggregateEventInterface) {
             $event->setIdentifier($this->getIdentifier());
         }
-
-        $messageMetadata = new MessageMetadata($metadata);
-
         $this->apply($event);
-
-        $this->events[] = new EventTransport($event, $messageMetadata);
+        $this->events[] = $event;
```

Comment:
I renamed it to `reconstitutedVersion` now and made the field private to avoid confusion